consumer/protocol/rpc_extensions.go

```diff
@@ -102,6 +102,8 @@ func (m *ApplyRequest_Change) Validate() error {
 			return pb.ExtendContext(err, "Delete")
 		} else if m.ExpectModRevision <= 0 && (m.ExpectModRevision != -1) {
 			return pb.NewValidationError("invalid ExpectModRevision (%d; expected > 0 or -1)", m.ExpectModRevision)
+		} else if m.PrimaryHints != nil {
```

Comment:
What about validation of the hints themselves? I'm not actually sure to what extent we really _can_ validate them, as long as they parse OK, especially in this function.